bin/mocha.js

```diff
@@ -109,7 +113,15 @@ if (mochaArgs['node-option'] || Object.keys(nodeArgs).length || hasInspect) {
   proc.on('exit', (code, signal) => {
     process.on('exit', () => {
       if (signal) {
-        process.kill(process.pid, signal);
+        const numericSignal =
+          typeof signal === 'string' ? os.constants.signals[signal] : signal;
+        if (mochaArgs['posix-exit-codes'] === true) {
+          process.exit(SIGNAL_OFFSET + numericSignal);
+        } else {
+          process.kill(process.pid, signal);
+        }
```

Comment:
If the `process.kill(process.pid, signal)` is important because it will trigger handles like the `process.on('SIGINT'`, then that's what always should be used here.

Else the `process.exit()` variant should always be used.

Though the logic of Mocha here is kind of weird – the `signal` comes from the child process, and the `SIGNAL_OFFSET + numericSignal` should be sent if the process itself, not one of its childs, gets sent the `signal` – so I would say that `process.exit(SIGNAL_OFFSET + numericSignal)` here is wrong and that the original `process.kill(process.pid, signal)` is weird (as its essentially bubbling up the process signal? Is that common practice?)